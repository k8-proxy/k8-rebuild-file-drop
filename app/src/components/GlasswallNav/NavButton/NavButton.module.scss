.button {
	position: relative;
	width: 100%;
	height: 4.4rem;
	border: none;
	border-bottom: hsla(255, 100%, 100%, 0.2) solid 1px;
	background: transparent;
	font-size: 1.6rem;
	color: #fff;
	text-align: center;
	transition: all 0.3s;
	outline: none;
	cursor: pointer;

	&:hover {
		background-color: hsla(0, 100%, 100%, 0.1);
	}
}

.selected {
	color: hsl(205, 74%, 18%);
	background-color: hsla(0, 0%, 100%, 0.6);

	&:hover {
		color: hsl(205, 74%, 18%);
		background-color: hsla(0, 0%, 100%, 0.6);
	}
}
