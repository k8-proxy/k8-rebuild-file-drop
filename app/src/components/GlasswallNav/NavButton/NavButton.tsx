import React from "react";
import styles from "./NavButton.module.scss";

export interface NavButtonProps { selected: boolean, clickHandler?: Function, children?: React.ReactNode }

function NavButton(props: NavButtonProps) {
    return (
        <li>
            <button
                className={`${styles.button} ${props.selected ? styles.selected : ""}`}
                onClick={() => props.clickHandler && props.clickHandler()}>{props.children}</button>
        </li>
    );
}

export { NavButton };