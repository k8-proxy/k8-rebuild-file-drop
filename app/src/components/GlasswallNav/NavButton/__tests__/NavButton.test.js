import React from "react";
import { shallow, mount } from "enzyme";
import TestRenderer from "react-test-renderer";
import { NavButton } from "../NavButton";

const selected = true;

const clickHandler = () => {
    console.log("click handled.")
};

const props = {
    selected: selected,
    clickHandler: clickHandler
};

test("NavButton_Snapshot", () => {
    // Arrange
    const component = TestRenderer.create(<NavButton {...props} />);

    // Act
    let tree = component.toJSON();

    // Assert
    expect(tree).toMatchSnapshot();
});

test("Displays_Correct_Props", () => {
    // Arrange
    // Act
    const navButtonComponent = mount(<NavButton {...props}/>);

    // Assert
    expect(navButtonComponent.prop("selected")).toEqual(props.selected);
    expect(navButtonComponent.prop("clickHandler")).toEqual(props.clickHandler);    
});