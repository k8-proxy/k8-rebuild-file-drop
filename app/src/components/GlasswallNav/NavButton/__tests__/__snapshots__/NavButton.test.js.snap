// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavButton_Snapshot 1`] = `
<li>
  <button
    className="button selected"
    onClick={[Function]}
  />
</li>
`;
